src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/queue_tuning/RcaItQueueTuning.java

```diff
@@ -0,0 +1,106 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.QueueRejectionClusterRca;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(RcaItNotEncryptedRunner.class)
+
+@AClusterType(ClusterType.MULTI_NODE_CO_LOCATED_MASTER)
+@ARcaGraph(ElasticSearchAnalysisGraph.class)
+//specify a custom rca.conf to set the rejection-time-period-in-seconds to 5s to reduce runtime
+@ARcaConf(dataNode = RcaItQueueTuning.QUEUE_TUNING_RESOURCES_DIR + "rca.conf")
+@AMetric(name = ThreadPool_RejectedReqs.class,
+    dimensionNames = {ThreadPoolDimension.Constants.TYPE_VALUE},
+    tables = {
+        @ATable(hostTag = HostTag.DATA_0,
+            tuple = {
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.WRITE_NAME},
+                    sum = 1.0, avg = 1.0, min = 1.0, max = 1.0),
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.SEARCH_NAME},
+                    sum = 0.0, avg = 0.0, min = 0.0, max = 0.0)
+            }
+        ),
+        @ATable(hostTag = {HostTag.ELECTED_MASTER},
+            tuple = {
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.WRITE_NAME},
+                    sum = 0.0, avg = 0.0, min = 0.0, max = 0.0),
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.SEARCH_NAME},
+                    sum = 0.0, avg = 0.0, min = 0.0, max = 0.0)
+            }
+        )
+    }
+)
+
+@AMetric(name = ThreadPool_QueueCapacity.class,
+    dimensionNames = {ThreadPoolDimension.Constants.TYPE_VALUE},
+    tables = {
+        @ATable(hostTag = HostTag.DATA_0,
+            tuple = {
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.WRITE_NAME},
+                    sum = 500, avg = 500, min = 500, max = 500),
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.SEARCH_NAME},
+                    sum = 1500, avg = 1500, min = 1500, max = 1500)
+            }
+        ),
+        @ATable(hostTag = {HostTag.ELECTED_MASTER},
+            tuple = {
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.WRITE_NAME},
+                    sum = 0.0, avg = 0.0, min = 0.0, max = 0.0),
+                @ATuple(dimensionValues = {ThreadPoolType.Constants.SEARCH_NAME},
+                    sum = 1500, avg = 1500, min = 1500, max = 1500)
+            }
+        )
+    }
+)
+public class RcaItQueueTuning {
+  public static final String QUEUE_TUNING_RESOURCES_DIR =
+      "./src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/queue_tuning/resource/";
+  private TestApi api;
+
+  @Test
+  @AExpect(
+      what = AExpect.Type.REST_API,
+      on = HostTag.ELECTED_MASTER,
+      validator = QueueRejectionValidator.class,
+      forRca = QueueRejectionClusterRca.class,
+      timeoutSeconds = 500)
+  public void simple() {
+  }
+
+  public void setTestApi(final TestApi api) {
+    this.api = api;
```

Comment:
`setTestApi()` is an optional method. You can define this, if you want access to the TestApi object. As in this case we are not using that, maybe we can skip this and the class member  `private TestApi api;` ?

Reply:
done. removed the testapi.